### ISSUE TITLE
`deno publish --dry-run` fails on class private function overloading

### ISSUE THREAD

**Issue by williamhorning** [denoland/deno#22910]
Version: Deno 1.41.2

When running `deno publish --dry-run` in one of my projects, the CLI throws an error when it encounters private class function overloading, giving a TS2300 (duplicate identifier) error.

logs:

```ts
Checking for slow types in the public API...
Check file:///home/jersey/projects/jsr-publish-repro/mod.ts
error: Failed ensuring public API type output is valid.

TS2300 [ERROR]: Duplicate identifier 'int_test'.
  private int_test!: unknown;
          ~~~~~~~~
    at file:///home/jersey/projects/jsr-publish-repro/mod.ts:7:17

TS2300 [ERROR]: Duplicate identifier 'int_test'.
  private int_test!: unknown;
          ~~~~~~~~
    at file:///home/jersey/projects/jsr-publish-repro/mod.ts:8:17

Found 2 errors.

You may have discovered a bug in Deno. Please open an issue at: https://github.com/denoland/deno/issues/
```

minimal repro:

```ts
export default class test {
  test_delay(): Promise<undefined> {
    return new Promise((resolve) => setTimeout(resolve, 1000));
  }

  private async int_test(a: true, b: undefined): Promise<boolean>;
  private async int_test(a: false, b: string): Promise<boolean>;
  private async int_test(a: boolean, b?: string): Promise<boolean> {
    if (b) console.log(b);

    await this.test_delay();

    return a;
  }
}
```

**Comment by williamhorning** [comment]
while doing some more testing, using `--allow-slow-types` gets around the error, though that isn't exactly ideal

**Comment by dsherret** [comment]
This was fixed in 1.42.3.
